### PR TITLE
HaversineIntermediate trait to find intermediate points along a great circle route

### PR DESCRIPTION
**Pull request by JivanRoquet**
Given two existing points, implements the ability to find intermediate points along the great circle route joining the initial points.

The `haversine_intermediate` method takes three parameters:

- `&self`
- `other: &Point<T>` which represents the other side of the great circle route
- `f: T` which represents the fraction of the route at which to put the intermediate point

`f` must be included between `T::zero()` and `T::one()`.

Return value is `Point<T>`.

### CODE REVIEW

File: geo/src/algorithm/haversine_intermediate.rs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        let d = two * k.asin();
+
+        let a = ((one - f) * d).sin() / d.sin();
+        let b = (f * d).sin() / d.sin();
+
+        let x = a * lat1.cos() * lon1.cos() + b * lat2.cos() * lon2.cos();
+        let y = a * lat1.cos() * lon1.sin() + b * lat2.cos() * lon2.sin();
+        let z = a * lat1.sin() + b * lat2.sin();
+
+        let dxy = (x.powi(2) + y.powi(2)).sqrt();
```
Comment (frewsxcv) [inline geo/src/algorithm/haversine_intermediate.rs]: I think this should be the same as `x.hypot(y)`

https://docs.rs/num/0.1.42/num/trait.Float.html#tymethod.hypot

Comment (JivanRoquet) [reply geo/src/algorithm/haversine_intermediate.rs]: agreed!

**Comment by JivanRoquet** [comment]
Added the ability to "fill" a great circle route with intermediate points at regular interval, as described in issue https://github.com/georust/rust-geo/issues/228.

**Comment by JivanRoquet** [comment]
up? :)

**Comment by frewsxcv** [comment]
lgtm, thanks!

bors r+

**Pull request merged**